superset/connectors/sqla/models.py

```diff
@@ -436,6 +436,25 @@ def get_from_clause(self, template_processor=None, db_engine_spec=None):
             return TextAsFrom(sa.text(from_sql), []).alias('expr_qry')
         return self.get_sqla_table()
 
+    def adhoc_metric_to_sa(self, metric):
+        column_name = metric.get('column').get('column_name')
+        aggregate = metric.get('aggregate')
+        label = metric.get('label')
+        if aggregate == 'COUNT_DISTINCT':
```

Comment:
nit: I would make this a `dict()` mapping for instance:
```
aggregate_dict = {
     'SUM' : sa.func.COUNT(column(column_name))
     ...
}
sa_metric = aggregate_dict.get(aggregate)
```

I think this is much cleaner and would allow others to add more metrics later.